code/modules/mechs/mech_interaction.dm

```diff
@@ -171,6 +171,9 @@
 		return attack_self(user)
 	else if(adj)
 		setClickCooldown(arms ? arms.action_delay : 15)
+		if(ismob(A))
+			var/mob/target = A
```

Comment:
Don't think you need this cast.

Reply:
Should be handled by the arg types in the proc call.